### DISCUSSION TITLE
Missing subpackages ?

### DISCUSSION THREAD

**Discussion by Melanie-devv** [MAIF/melusine#180] [Q&A] [answered]
Hi Melusine team,

First of all, thank you for your work. 
I'm encountering an issue with some imports that worked in older versions but seem to no longer be available in the current version.

I cloned the repository and installed Melusine with pip install melusine. However, I'm unable to do :
`from melusine.summarizer.keywords_generator import KeywordsGenerator`

The summarizer module is marked as a "missing import," and I can't find the summarizer/keywords_generator.py file (nor the prepare_email, nlp_tools,... folders) in the current version of the repository.

I wanted to know if this is an intentional change, or if these files were moved, renamed, deleted in a recent update ?

Thank you in advance for your help and clarification

**Comment by HugoPerrier** [comment, marked as answer]
Hello Melanie and thank you for your contribution.

Melusine 3.0.0 is a full rewrite of the code with massive breaking changes.
In the short term, if you need to make your existing code work, you can downgrade to version 2.3.6

Otherwise I recommend to migrate towards the new version, here are a few contents you may find useful:
- New documentation https://maif.github.io/melusine/
- Getting started tutorial https://maif.github.io/melusine/tutorials/00_GettingStarted/
- Article describing why we went for a full rewrtie of Melusine https://medium.com/oss-by-maif/rewriting-a-python-package-the-journey-towards-melusine-v3-0-c57e2b6cee45

Long story short, the new version favors stability and maintenance over multiplying features.

- summarizer => With the rise of GenAI and Open Source models available on HuggingFace we decided to drop this feature
If you want to re-create the KeywordsGenerator in your project, the code is available here:
https://github.com/MAIF/melusine/commit/2c4c5266e05e004325aef64e3353446b63130465#diff-78d59062be371fe1809f0b61420d994c510cd3a6be9123c52ccbe002957f595eL47

The features of the prepare_email subpackage are mostly migrated in the new Melusine (check out the preprocessing_pipeline)
The features of the nlp_tools subpackage are either dropped (many were just wrappers over sklearn or Gensim classes) or re-implemented (you might be interested in the processors.py file)

Hope it helps, let me know if you have further questions

**Reply by Melanie-devv** [reply]
Okay I see, thanks !👍
